### PR TITLE
Milestone: Private Events

### PR DESCRIPTION
**Pull request by enkog**
In this project, I built a site similar to a private Eventbrite which allows users to create events and then manage user signups. Users can create and attend many events.
The aim of the project is to master associations in rails.

### CODE REVIEW

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @enkog 
Such a remarkable project you made :+1: ! Awesome README :100: !
We, however, do have a few changes to carry out before we proceed (see review comment).

##

 For this project, we have to make sure nothing is messed up. Kindly compare the screenshots below 👇 and work on the suggested changes.
- [x] Event's index page is not displaying past and upcoming events list please fix it. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] There is an issue after signup and login with event controller due to unknown user_id. Please fix this issue ([screenshot](https://user-images.githubusercontent.com/29689273/108851735-949b0200-7606-11eb-926b-f49707fca2cb.png))
- [x] There is an issue  due to unknown user_id. Please fix this issue ([Screenshot](https://user-images.githubusercontent.com/29689273/108851728-92d13e80-7606-11eb-90eb-4d059071e592.png))

##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Comment by enkog** [comment]
Hello @umairarshadbutt,

Thank you for your feedback. Kindly note that the user_id field was part of a previous and not the most recent migration. However, I have deleted the migration file containing that field definition so you would not get the error you got the last time. Also, I have already shown the past and upcoming events and I am surprised that you could not see that on your end. 


Please find the screenshots attached showing these pages.

Kind Regards

upcoming events

past events

user page

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @enkog  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] When I go to the user show page and creating event => the events index page, There errors which are thrown and make the app to crash, learn the following screenshots to know about them
Screenshot from 2021-02-24 10-16-46
Screenshot from 2021-02-24 10-16-56

___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @enkog    ! :wave: :wave:

#### You have done well so far :clap: However, there are changes required for you to make in order for this project to be approved. Kindly work on them and submit them for another code review.


**Recommended**
- [x] I, unfortunately,  could not run this app, as I got an error upon sign up.

Screenshot (99)

- [x] You have some rubocop errors as well. Run `rubocop -a` in your terminal to fix most of them.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

### Hello @enkog please do not feel discouraged by the requirements because this is a learning process and what we all want is for you to get the best understanding/implementation of this project.
You can find the required changes below:
- [x] After I sign up in your app I get the following error screen
- [x] When I try to create an event I get the following error 
screenn

### **Please make the required changes and submit another review request when you are ready.**

<hr>
### if you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Status: Changes Required
### Hello again @enkog, good job implementing last changes, you are almost there. I am going to ask you to do few more changes which I could not see before because I could not run your app
## Changes Required general:
- [x] Fix rubocop offenses locally, currently you have 36 offenses. Run `rubocop -a` to fix most of them
- [x]  The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

### **UsersController** _`show` action_:
- [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

### **Please make the required changes and submit another review request when you are ready.**

<hr>
If you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @enkog    ! :wave: :wave:

#### You have worked very hard on this project so far :clap: and while I would've loved to approve this project, there is still one change required for you to make in order for this project to be approved. 


**Recommended**

- [x] You have some rubocop errors as well. Run `rubocop -a` in your terminal to fix most of them.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Comment by enkog** [comment]
Below is a screenshot of my local machine with no rubocop errors.

terminal

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hello Team :wave:
Status: Changes Required :recycle:

Good job. I like your design, your app is functional. Your repo looks professional
But before approving this project let's cover a few details together to make your project better :blush: :arrow_double_down: .

REQUIRED:

- [x] In your UsersController#show you should: lists events created by user, i.e. @created_events = current_user.events ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [ ] In your `events table` : an `index` for `creator_id` column is added (It is important because, queries on this column will be faster with indexes)
- [ ] You have to put all logics in your view inside helper.


- [ ] Models have unit tests for validations and associations (`Rspec`) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))*
- [ ] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [ ] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

**To sum up:**
- [ ] Use helper.
- [ ] Add index for creator_id.
- [ ] Add test units using Rspec + Capybara
- [ ] In your Readme/Setup , before `$ rails db:migrate`, user must create the DB by `$ rails db:create`

That's it, you can submit another code review request after making these changes
With all my respect :pray:

Rindra Josia

**Comment by mimipeshy** [comment]
### STATUS :APPROVED :partying_face: 

Hey there @enkog  :wave: :wave:  good job on working on the requested changes. This project is therefore approved. Go ahead and merge. All the best in the next milestones :man_technologist: 

Regards
Peris Ndanu
